src/main/java/Duke.java

```diff
@@ -1,10 +1,193 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.SimpleTimeZone;
+
 public class Duke {
-    public static void main(String[] args) {
+    protected final TaskList taskList;
+    protected final ArrayList<String> commandList;
+    protected Parser parser;
+    protected Storage storage;
+    protected Ui ui;
+    protected String directoryPath;
+    protected String dataFilePath;
+
+    protected Duke() {
+        this.dataFilePath = "./data/duke.txt";
+        this.directoryPath = "./data";
+        this.commandList = new ArrayList<>();
+        this.taskList = new TaskList();
+        this.parser = new Parser();
+        this.storage = new Storage(directoryPath, dataFilePath);
+        this.ui = new Ui();
+    }
+
+    // methods
+    public void addTask(Task task_given) {
+        this.taskList.add(task_given);
+    }
+
+    public String msg_of_add(Task task) {
+        return "Got it. I've added this task:\n " + task.toString() + "\nNow you have " +
+                taskList.size() + " tasks in the list:D";
+    }
+
+    public String print_curr_taskList() {
+        return taskList.print_curr_tasks();
+    }
+
+
+    public String mark_as_done(int index) throws DukeException {
+        if (taskList.size() <= index) {
+            throw new WrongIndexException();
+        }
+        Task task = this.taskList.get(index);
+        task.mark();
+        return "Nice! I've marked this task as done:\n" + task.toString();
+    }
+
+    public String mark_as_undone(int index) throws DukeException {
+        if (taskList.size() <= index) {
+            throw new WrongIndexException();
+        }
+        Task task = this.taskList.get(index);
+        task.unmark();
+        return "OK, I've marked this task as not done yet:\n" + task.toString();
+    }
+
+    public String delete_msg(int index) throws WrongIndexException {
+        if (taskList.size() <= index) {
+            throw new WrongIndexException();
+        }
+        Task task = taskList.get(index);
+        taskList.remove(index);
+        return "Noted. I've removed this task:\n " + task.toString() +
+                "\nNow you have " + taskList.size() + " tasks in the list.";
+
+    }
+
+    public void loadDataFromDisk() {
```

Comment:
Perhaps this method could be added under Storage class instead of Duke? 🤔 Especially since loading data can be considered to be an action under storage.